### PR TITLE
Fix/rework tx build

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

I'm changing how tx parameters are passed to the `build_transaction` function, and merging that one with `_build_transaction` since now it doesn't make sense to have two of those, but we'll see once I finish the matching PR in consensus-algorithms. 

## Fixes

Changes build_transaction and merges it with _build_transaction.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`scripts/generate_ipfs_hashes.py`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py [hunk 1]
```diff
@@ -1246,57 +1246,27 @@ def build_transaction(  # pylint: disable=too-many-arguments
...
         tx = method(**method_args)
-        tx = self._build_transaction(
-            tx,
-            tx_args["sender_address"],
-            tx_args["eth_value"],
-            tx_args["gas"],
-            tx_args["gas_price"],
-            tx_args["max_fee_per_gas"],
-            tx_args["max_priority_fee_per_gas"],
-        )
```
Comment (dvilelaf) [inline plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py:1257]: I had to change this as obviously some of these fields might be missing and we have not yet checked for them being present at this stage.

File: plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py [hunk 2]
```diff
@@ -1246,57 +1246,27 @@ def build_transaction(  # pylint: disable=too-many-arguments
...
-        :param max_fee_per_gas: the transaction max_fee_per_gas
-        :param max_priority_fee_per_gas: the transaction max_priority_fee_per_gas
-        :return: the final transaction
-        """
-        nonce = self.api.eth.get_transaction_count(sender_address)
+        nonce = self.api.eth.get_transaction_count(tx_args["sender_address"])
         tx_params = {
             "nonce": nonce,
-            "value": eth_value,
+            "value": tx_args["eth_value"] if "eth_value" in tx_args else 0,
```
Comment (DavidMinarsch) [inline plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py:1253]: Generally speaking I'd change this to `value` here as this is also the web3 terminology. And over here we are closer to that then to our other repo's semantics.

File: plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py [hunk 3]
```diff
@@ -1246,57 +1246,27 @@ def build_transaction(  # pylint: disable=too-many-arguments
...
+
         if (
-            gas_price is None
-            and max_fee_per_gas is None
-            and max_priority_fee_per_gas is None
+            "gasPrice" not in tx_params
+            and "maxFeePerGas" not in tx_params
+            and "maxPriorityFeePerGas" not in tx_params
         ):
             gas_data = self.try_get_gas_pricing()
```
Comment (DavidMinarsch) [inline plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py:1271]: Here you need to pass along `old_tip` if it is present in the tx arguments

[the remark this answers is not in the document]
Comment (DavidMinarsch) [reply plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py:1271]: The problem almost certainly is caused elsewhere then.

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**